gossipd/gossip.c

```diff
@@ -1817,17 +1821,24 @@ static void try_reach_peer(struct daemon *daemon, const struct pubkey *id,
 /* Called from timer, so needs single-arg declaration */
 static void retry_important(struct important_peerid *imp)
 {
+	/* In case we've come off a timer, don't leave dangling pointer */
+	imp->reconnect_timer = NULL;
 	try_reach_peer(imp->daemon, &imp->id, false);
 }
 
 static struct io_plan *connect_to_peer(struct io_conn *conn,
 				       struct daemon *daemon, const u8 *msg)
 {
 	struct pubkey id;
+	struct important_peerid *imp;
 
 	if (!fromwire_gossipctl_connect_to_peer(msg, &id))
 		master_badmsg(WIRE_GOSSIPCTL_CONNECT_TO_PEER, msg);
 
+	/* If this is an important peer, free any outstanding timer */
+	imp = important_peerid_map_get(&daemon->important_peerids, &id);
+	if (imp)
+		imp->reconnect_timer = tal_free(imp->reconnect_timer);
```

Comment:
At first glance `imp->reconnect_timer` seems not initialized when new `important_peerid` is created.  However `important_peerid` is created only one place, and, it calls `retry_important` immediately, which sets `imp->reconnect_timer` to `NULL`, so no issue here.